### PR TITLE
CI setup 

### PR DESCRIPTION
**Pull request by LennardBoediger**
all test passed

### CODE REVIEW

**Comment by LennardBoediger** [comment]
ready to merge

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+    - if: '$CI_PIPELINE_SOURCE == "web"'
+  tags:
+    - docker
+  stage: build
+  script:
+    - make -C runners/lpc55 objcopy BOARD=nk3xn
+  after_script:
+    - cp ./runners/lpc55/firmware-*.bin artifacts
```
Comment (robin-nitrokey) [inline .gitlab-ci.yml:105]: Can we `mkdir -p artifacts` instead of adding `artifacts/.keep` to version control?

**Comment by robin-nitrokey** [comment]
[Merging the `make` steps into a single job](https://github.com/Nitrokey/solo2/commit/67f5237bd5d84590c1c44e2ec0637ecac2e0c098) (per board) reduces the execution time by half.  Caching `runners/lpc55/target` and `$CARGO_HOME` = `/usr/local/cargo` should further speed up the process, but I didn’t manage to set that up.

**Comment by LennardBoediger** [comment]
> [Merging the `make` steps into a single job](https://github.com/Nitrokey/solo2/commit/67f5237bd5d84590c1c44e2ec0637ecac2e0c098) (per board) reduces the execution time by half. Caching `runners/lpc55/target` and `$CARGO_HOME` = `/usr/local/cargo` should further speed up the process, but I didn’t manage to set that up. Alternatively, [running them sequentially](https://github.com/Nitrokey/solo2/commit/fa3b2f7fa17d7c7258bc2e2e73c4484d2ea8c9fc) speeds up the process slightly too.

Sequential jobs were turned off for troubleshooting reasons. 
I turned it back on, the resulting build time for 6 separate jobs is 02:42. (https://git.dotplex.com/nitrokey/solo2/-/pipelines/13225)
So it's just as fast as #fa3b2f7f, but allows a clean overview of which builds were successful

File: .gitlab-ci.yml (hunk 1 above)
Comment (LennardBoediger) [reply .gitlab-ci.yml]: done

**Comment by LennardBoediger** [comment]
all test passed
pull: https://git.dotplex.com/nitrokey/solo2/-/pipelines/13507
build: https://git.dotplex.com/nitrokey/solo2/-/pipelines/13509
Nighly: https://git.dotplex.com/nitrokey/solo2/-/pipelines/13510

**Review by szszszsz — CHANGES_REQUESTED** [review]
1. Please make it working locally
2. Would be nice to have a single `CI` recipe, which would be called by the CI, and would allow to test the execution locally without looking into the yaml file

**Comment by szszszsz** [comment]
My output while running locally:
```
$ make -f docker-Makefile docker-build BOARD=nk3xn 
docker run -i --rm -v work/solo2:/app nitrokey3 make -C runners/lpc55 build BOARD=nk3xn           
make: *** runners/lpc55: No such file or directory.  Stop.                                              
make: *** [docker-Makefile:7: docker-build] Error 2                                                     
```

**Comment by LennardBoediger** [comment]
it works locally for me.
created CI recipe in the Makefile

**Review by szszszsz — APPROVED** [review]
LGTM!

**Comment by szszszsz** [comment]
Merged! I have changed the commit message hence it is not recognized by Github as so, thus the "closed" message.

**Pull request closed without merging**